tests/fast-integration/skr-svcat-migration-test/test-helpers.js

```diff
@@ -63,6 +69,85 @@ async function installBTPServiceOperatorMigrationHelmChart() {
     }
 }
 
+async function getFunctionPod(functionName) {
+    let labelSelector = `serverless.kyma-project.io/function-name=${functionName},serverless.kyma-project.io/resource=deployment`;
+    let res = {};
+    for (let i = 0; i < 30; i++) {
+        res = await listPods(labelSelector);
+        if (res.body.items.length == 1) {
+            let pod = res.body.items[0];
+            if (pod.status.phase == "Running") {
+                return pod
+            }
+        }
+        sleep(10000);
+    }
+    if (res.body.items.length != 1) {
+        let podNames = res.body.items.map(p=>p.metadata.name);
+        let phases = res.body.items.map(p=>p.status.phase);
+        throw new Error(`Failed to find function ${functionName} pod in 5 minutes. Expected 1 ${labelSelector} pod with phase "Runninng" but found ${res.body.items.length}, ${podNames}, ${phases}`);
+    }
+}
+
+async function checkPodPresetEnvInjected() {
+    let cmd = `for v in {vars}; do x="$(eval echo \\$$v)"; if [[ -z "$x" ]]; then echo missing $v env variable; exit 1; else echo found $v env variable; fi; done`;
+    for (let f of functions) {
+        let pod = await getFunctionPod(f.name);
+        let envCmd = cmd.replace("{vars}",f.checkEnvVars);
+        await kubectlExecInPod(pod.metadata.name, "function", ["sh", "-c", envCmd]);
+    }
+}
+
+async function restartFunctionsPods() {
+    let podNames = {}
+    for (let f of functions) {
+        let pod = await getFunctionPod(f.name);
+        console.log("delete pod", pod.metadata.name);
+        await deleteK8sResource(pod);
+        podNames[f.name] = pod.metadata.name;
+    }
+
+    let finished = false;
+    let needsPoll = [];
+    for (let i = 0; i < 10; i++) {
+        needsPoll = [];
+        for (let f of functions) {
```

Comment:
Interesting syntax - I'm still used to `array.forEach()`.

Reply:
`forEach` is my preferred too, and it works great unless the functions inside need to be `async` because they leverage `await`. In that case `forEach` loops over and gives you bunch of promises. `for (let x of arr)` allows you to call `await` and they will really wait :)

Reply:
Ah! Today I learned... :)